File: surf-friends/src/main/java/dev/slne/surf/friends/FriendData.java
package dev.slne.surf.friends;

import it.unimi.dsi.fastutil.objects.ObjectList;
import java.util.UUID;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Setter
@Getter
public class FriendData {
  private UUID player;
  private ObjectList<UUID> friends;
  private ObjectList<UUID> friendRequests;
  private Boolean allowRequests;
}

File: surf-friends/src/main/java/dev/slne/surf/friends/FriendData.kt
package dev.slne.surf.friends

import it.unimi.dsi.fastutil.objects.ObjectArraySet
import it.unimi.dsi.fastutil.objects.ObjectSet
import java.util.*

data class FriendData(
    val player: UUID,
    val friends: ObjectSet<UUID>  = ObjectArraySet(),
    val friendRequests: ObjectSet<UUID> = ObjectArraySet(),
    var allowRequests: Boolean = true
)
